tests/e2e/e2e_test.go

```diff
@@ -256,29 +358,410 @@ func (s *IntegrationTestSuite) TestConcentratedLiquidity() {
 		s.Require().Equal(position.UpperTick, int64(upperTick))
 	}
 
-	// assert positions for address1
+	// Assert positions for address1
 	addr1position1 := positionsAddress1[0]
 	addr1position2 := positionsAddress1[1]
-	// first position first address
+	// First position first address
 	validateCLPosition(addr1position1, poolID, -1200, 400)
-	// second position second address
-	validateCLPosition(addr1position2, poolID, -400, 400)
+	// Second position second address
+	validateCLPosition(addr1position2, poolID, -400, 1200)
 
-	// assert positions for address2
+	// Assert positions for address2
 	addr2position1 := positionsAddress2[0]
-	// first position second address
+	// First position second address
 	validateCLPosition(addr2position1, poolID, 2200, maxTick)
 
-	// assert positions for address3
+	// Assert positions for address3
 	addr3position1 := positionsAddress3[0]
 	addr3position2 := positionsAddress3[1]
-	// first position third address
+	// First position third address
 	validateCLPosition(addr3position1, poolID, -1600, -200)
-	// second position third address
+	// Second position third address
 	validateCLPosition(addr3position2, poolID, minTick, 1400)
 
-	// Withdraw Position:
+	// Collect Fees
+
 	var (
+		// feeGrowthGlobal is a variable for tracking global fee growth
+		feeGrowthGlobal = sdk.ZeroDec()
+		outMinAmt       = "1"
+	)
+
+	// Swap 1
+	// Not crossing initialized ticks => performed in one swap step
+	// Swap affects 3 positions: both that address1 has and one of address3's positions
+	// Asserts that fees are correctly collected for non cross-tick swaps
+	var (
+		// Swap parameters
+		uosmoInDec_Swap1 = sdk.NewDec(3465198) // Minus swap fee = 3430546.02
+		uosmoIn_Swap1    = fmt.Sprintf("%suosmo", uosmoInDec_Swap1.String())
+	)
+	// Perform swap (not crossing initialized ticks)
+	node.SwapExactAmountIn(uosmoIn_Swap1, outMinAmt, fmt.Sprintf("%d", poolID), denom0, initialization.ValidatorWalletName)
+	// Calculate and track global fee growth for swap 1
+	feeGrowthGlobal.AddMut(calculateFeeGrowthGlobal(uosmoInDec_Swap1, swapFeeDec, concentratedPool.GetLiquidity()))
+
+	// Let the chain pick up the changes:
+	chainA.WaitForNumHeights(2)
```

Comment:
Is this necessary? Swaps are executed with `-b=block` so changes should be propagated immediately. Let's remove delays that we don't need to avoid increasing CI times